examples/summarizePatches.py

```diff
@@ -0,0 +1,179 @@
+                if patchBBox.overlaps(skymapBBoxes[skymapId]):
+                    visitId, ccdId = skymapId
+                    if visitId not in visitsToCcds:
+                        visitsToCcds[visitId] = [ccdId]
+                        visitFilter = butler.queryMetadata('calexp_md', 'filter', visit=visitId)[0]
+                        if visitFilter not in filtersToVisits:
+                            filtersToVisits[visitFilter] = []
+                        filtersToVisits[visitFilter] += [visitId]
+                        numOverlaps += 1
+                    else:
+                        visitsToCcds[visitId] += [ccdId]
+                        numOverlaps += 1
+
+            # If there was an overlap, print and save information
+            if numOverlaps != 0:
+                print("patch=%r:    Found %r overlapping ccds." % (patchId, numOverlaps))
+
+                if saveOverview:
+                    overview.write("patch=" + str(patchId) + "\n")
+                    overview.write("    " + str(filtersToVisits) + "\n")
+
+                if saveSummaries is not None:
+                    for i_f, patchFilter in enumerate(filtersToVisits):
+
+                        # Write expanded file - one line for each filter in each patch
+                        expandedSummary.write("filter=" + patchFilter + " ")
+                        expandedSummary.write("tract=" + str(tract.getId()) + " ")
+                        expandedSummary.write("patch=")
+                        expandedSummary.write(str(patchId[0]) + "," + str(patchId[1]) + " ")
+                        for i_v, visit in enumerate(filtersToVisits[patchFilter]):
+                            if i_v != 0:
+                                expandedSummary.write("^" + str(visit))
+                            else:
+                                expandedSummary.write("--selectId visit=" + str(visit))
+                        expandedSummary.write("\n")
+
+                        # Write condensed file - one line for each patch
+                        if i_f != 0:
+                            condensedSummary.write("^" + patchFilter)
+                        else:
+                            condensedSummary.write("filter=" + patchFilter)
+                    condensedSummary.write(" tract=" + str(tract.getId()) + " ")
+                    condensedSummary.write("patch=")
+                    condensedSummary.write(str(patchId[0]) + "," + str(patchId[1]) + " ")
+                    condensedSummary.write("\n")
+
+                # Save an overview of ccd/patch overlaps
+                if saveOverview:
+                    for visit in visitsToCcds:
+                        overview.write("        visit=" + str(visit) + "\n")
+                        for ccd in visitsToCcds[visit]:
+                            overview.write("            ccd=" + str(ccd) + "\n")
+                    overview.write("\n")
+
+def splitId(argName):
+    class SplitIdValueAction(IdValueAction):
```

Comment:
Yes, a future work of this can be to make it a CmdLineTask. But DM-5766 does look much better; thanks for mentioning it. And thanks for mentioning other relevant code too. I think we looked at some but probably not all of them.